src/libopensc/card-starcos.c

```diff
@@ -1713,17 +1792,37 @@ static int starcos_compute_signature(sc_card_t *card,
 		SC_FUNC_RETURN(card->ctx, SC_LOG_DEBUG_VERBOSE, SC_ERROR_INVALID_ARGUMENTS);
 
 	if (ex_data->sec_ops == SC_SEC_OPERATION_SIGN) {
+		u8 rbuf_ext[SC_MAX_EXT_APDU_BUFFER_SIZE];
 		/* compute signature with the COMPUTE SIGNATURE command */
 		
 		if (card->type == SC_CARD_TYPE_STARCOS_V3_4
 				|| card->type == SC_CARD_TYPE_STARCOS_V3_5) {
 			size_t tmp_len;
+			u8 apdu_case;
+			size_t apdu_le;
+			size_t apdu_resplen;
+			u8 * apdu_resp;
+
+			if (card->caps & SC_CARD_CAP_APDU_EXT
+				&& card->reader->max_recv_size > SC_MAX_APDU_BUFFER_SIZE
+				&& outlen > SC_MAX_APDU_BUFFER_SIZE ) {
```

Comment:
The GET RESPONSE command is not included in the 3.4 nor in the 3.5 manual, so I suppose it is not supported, but I can check it. This patch was intended for the v3.5 cards with 3072 bit RSA key. In this case the answer to COMPUTE DIGITAL SIGNATURE would exceed 256 bytes, and it would only work with extended APDU case 4. 
I will double check if the card works with SC_APDU_FLAGS_CHAINING.

Reply:
Chaining is for the command, not the response. I just wanted to make you aware of the problem that some readers may not be able to send/receive your data.

If there is no workaround for 3072 bit RSA with an only short-length-APDU-capable reader, then you should patch the code to assume that the reader is capable of sending extended length, as done in the [CardOS driver](https://github.com/OpenSC/OpenSC/blob/929717b505c8715cc97d0b44c965e09bb785a2d8/src/libopensc/card-cardos.c#L1199-L1254), for example. Unfortunately, the reader's capabilities cannot be reliably detected, so it's better to assume all requirements are met rather than failing without even trying.

Reply:
did you have the chance to look at the cardos link?

Reply:
Sorry for the delay and yes, I have already implemented a function to probe the reader possibilities, similar to the CardOS driver. Unfortunately I was quite busy in the past weeks, hopefully I will find time on the weekend to rebase the branch and test it through.